src/app/routes/home/getInitialData/index.js

```diff
@@ -17,42 +16,30 @@ const transformJson = pipe(
   squashTopStories,
   filterGroupsWithoutStraplines,
 );
+const getRadioScheduleToggle = path(['frontPageRadioSchedule', 'enabled']);
+const getRadioSchedulePosition = path(['frontPageRadioSchedule', 'value']);
 
-export const hasRadioSchedule = async (service, variant) => {
-  const config = await getConfig(service, variant);
-
-  const serviceHasRadioSchedule = pathOr(
-    false,
-    ['radioSchedule', 'hasRadioSchedule'],
-    config,
-  );
-
-  const radioScheduleOnFrontPage = pathOr(
-    false,
-    ['radioSchedule', 'onFrontPage'],
-    config,
-  );
-
-  return serviceHasRadioSchedule && radioScheduleOnFrontPage;
-};
-
-export default async ({ path, service, variant, pageType }) => {
+export default async ({ path: pathname, service, pageType, toggles }) => {
   try {
-    const pageHasRadioSchedule = await hasRadioSchedule(service, variant);
-    const pageDataPromise = fetchPageData({ path, pageType });
+    const pageDataPromise = fetchPageData({ path: pathname, pageType });
+    const radioScheduleIsEnabled = getRadioScheduleToggle(toggles);
+    const radioSchedulePosition = getRadioSchedulePosition(toggles);
```

Comment:
Not against this as such but is there any reason you get this radio schedule toggle info in `getInitialData` rather than in the `FrontPage` component itself? I'm torn between thinking this is nice because the presentation component (`FrontPage`) does not have any of the toggle concerns in it but also from the `FrontPage` component I found it weird to be getting the info about the radio schedule from the `pageData`! I think this approach is better but I wonder if we should be more consistent about doing this e.g. for ads: https://github.com/bbc/simorgh/blob/latest/src/app/pages/FrontPage/FrontPage.jsx#L100

Reply:
I had to get the toggle value here because it's used to determine whether to fetch radio schedule data on line 28 of this file. Any data fetching outside of this function doesn't guarantee SSR.

Reply:
Oh do radio schedules have to be SSR'ed? Interesting another barrier to clientside navigation 😞 Approach makes sense though